internal/idgen/idgen.go

```diff
@@ -18,36 +18,25 @@
 package idgen
 
 import (
-	"math/rand"
-	"sync"
-	"time"
-)
+	"strings"
 
-var (
-	seededIDGen  = rand.New(rand.NewSource(time.Now().UnixNano()))
-	seededIDLock sync.Mutex
+	"github.com/google/uuid"
 )
 
-func generateID() int64 {
-	seededIDLock.Lock()
-	defer seededIDLock.Unlock()
-	return seededIDGen.Int63()
-}
-
-// GenerateGlobalID generates global unique id
-func GenerateGlobalID() []int64 {
-	return []int64{
-		time.Now().UnixNano(),
-		0,
-		generateID(),
+// UUID generate UUID
+func UUID() (string, error) {
+	id, err := uuid.NewUUID()
+	if err != nil {
+		return "", err
 	}
+	return strings.ReplaceAll(id.String(), "-", ""), nil
 }
 
-// GenerateScopedGlobalID generates global unique id with a scopeId prefix
-func GenerateScopedGlobalID(scopeID int64) []int64 {
-	return []int64{
-		scopeID,
-		time.Now().UnixNano(),
-		generateID(),
+// GenerateGlobalID generates global unique id
+func GenerateGlobalID() string {
+	globalID, err := UUID()
+	if err != nil {
+		panic(err)
```

Comment:
When could cause an error? `panic` should be very serious as we are only an APM agent.

Reply:
I checked the source code for `github.com/google/uuid` and the error should always be nil. Anyway, I've now removed `panic`.